c/tskit/tables.h

```diff
@@ -1930,6 +2095,54 @@ on and may change arbitrarily between versions.
 */
 void tsk_site_table_print_state(const tsk_site_table_t *self, FILE *out);
 
+/**
+@brief Replace this table's data by copying from a set of column arrays
+
+@rst
+Clears the data columns of this table and then copies column data from the specified
+set of arrays. The supplied arrays should all contain data on the same number of rows.
+The metadata schema is not affected.
+@endrst
+
+@param self A pointer to a tsk_site_table_t object.
+@param num_rows The number of rows to copy from the specifed arrays.
+@param position The array of double position values to be copied.
+@param ancestral_state The array of char ancestral state values to be copied.
+@param ancestral_state_offset The array of tsk_size_t ancestral state offset values to be
+copied.
+@param metadata The array of char metadata values to be copied.
+@param metadata_offset The array of tsk_size_t metadata offset values to be copied.
+@return Return 0 on success or a negative value on failure.
+*/
+int tsk_site_table_set_columns(tsk_site_table_t *self, tsk_size_t num_rows,
+    const double *position, const char *ancestral_state,
+    const tsk_size_t *ancestral_state_offset, const char *metadata,
+    const tsk_size_t *metadata_offset);
+
+/**
+@brief Extends this table by copying from a set of column arrays
+
+@rst
+Copies column data from the specified set of arrays to create new rows at the end of the
+table. The supplied arrays should all contain data on the same number of rows. The
+metadata schema is not affected.
+@endrst
+
+@param self A pointer to a tsk_site_table_t object.
+@param num_rows The number of rows to copy from the specifed arrays.
+@param position The array of double position values to be copied.
+@param ancestral_state The array of char ancestral state values to be copied.
+@param ancestral_state_offset The array of tsk_size_t ancestral state offset values to be
```

Comment:
maybe indent the "copied" line here? Looks like doxygen doesn't care, so entirely optional

Reply:
Fixed it there and a few other places.